### DISCUSSION TITLE
Really cool project! - Larger example of third part library

### DISCUSSION THREAD

**Discussion by radderz** [IEvangelist/blazorators#27] [Ideas]
Just like in #14 I am interested in if it was possible to create a service implementation of a type script library.

It may not be a valid option but I would like to try this with something like a mapping library such as HERE Maps javascript API, Mapbox, bing maps, leaflet.js.

These APIs have objects that are created for the map related to the div you turn into the map usually, so there is a reference to that object to maintain. It'd be cool to start with the HERE Maps API which is written in typescript. Currently I am using this library in an old javascript API and would love to migrate it without having to write raw javascript interop.

Have you tried running this code gen against a complex API with many types and interfaces/services like this that also requires the maintenance of a reference object?

**Comment by IEvangelist** [comment]
Hi @radderz - thanks you for posting this question. As of right now, this project is limited and a bit prescriptive. I have not tried it on a large API yet, in fact it's been very narrowly scoped. I'm hoping that this project will be able to consume another OSS project that I'm working on - that will be able to parse any TypeScript type declaration. I'm re-writing the TypeScript compiler's parser, so that we'd be able to parse the abstract syntax tree into .NET. From this we'd have a much better capability to emit the desired C#.

As it stands today, this is manually parsed out using `Regex` and iterating line-by-line. It's a brute-force approach that was some of the remains from the initial hackathon. Does that make sense? If you're able and willing, I'd love to have contributors helping out with that project as well.
